packages/web-forms/src/components/common/map/useMapBlock.ts

```diff
@@ -0,0 +1,258 @@
+			featuresVectorLayer.value.dispose();
+			featuresVectorLayer.value = undefined;
+		}
+
+		const features = new GeoJSON().readFeatures(geoJSON, {
+			dataProjection: DEFAULT_GEOJSON_PROJECTION, // ToDo: do we support different GeoJSON projections?
+			featureProjection: mapInstance.value.getView().getProjection(),
+		});
+
+		features.forEach((feature) => {
+			if (feature.get(FEATURE_ID_PROPERTY) == null) {
+				feature.set(FEATURE_ID_PROPERTY, crypto.randomUUID());
+			}
+		});
+
+		featuresVectorLayer.value = new WebGLVectorLayer({
+			source: new VectorSource({ features }),
+			style: [
+				...getUnselectedStyles(FEATURE_ID_PROPERTY, SELECTED_ID_PROPERTY, SAVED_ID_PROPERTY),
+				...getSelectedStyles(FEATURE_ID_PROPERTY, SELECTED_ID_PROPERTY),
+				...getSavedStyles(FEATURE_ID_PROPERTY, SAVED_ID_PROPERTY),
+			],
+			variables: { [SAVED_ID_PROPERTY]: '', [SELECTED_ID_PROPERTY]: '' },
+		});
+		mapInstance.value.addLayer(featuresVectorLayer.value);
+		mapInstance.value.on('singleclick', (event) => selectFeatureByPosition(event.pixel));
+
+		currentState.value = STATES.READY;
+	};
+
+	const selectFeatureByPosition = (position: Pixel): void => {
+		const hitFeatures = mapInstance.value?.getFeaturesAtPixel(position, {
+			hitTolerance: MAP_HIT_TOLERANCE,
+			layerFilter: (layer) => layer === featuresVectorLayer.value,
+		});
+
+		const featureToSelect = hitFeatures?.length
+			? (hitFeatures[0] as Feature<GeometryType>)
+			: undefined;
+
+		selectFeature(featureToSelect);
+	};
+
+	const selectFeature = (feature: Feature<GeometryType> | undefined): void => {
+		selectedFeature.value = feature;
+
+		featuresVectorLayer.value?.updateStyleVariables({
+			[SELECTED_ID_PROPERTY]: (selectedFeature.value?.get(FEATURE_ID_PROPERTY) as string) ?? '',
+		});
+
+		if (selectedFeature.value != null) {
+			centerFeatureLocation(selectedFeature.value);
+		}
+	};
+
+	const unselectFeature = (): void => {
+		selectedFeature.value = undefined;
+		featuresVectorLayer.value?.updateStyleVariables({ [SELECTED_ID_PROPERTY]: '' });
+	};
+
+	const saveFeature = (): void => {
+		savedFeature.value = selectedFeature.value;
+		const savedFeatureId = (savedFeature.value?.get(FEATURE_ID_PROPERTY) as string) ?? '';
+		featuresVectorLayer.value?.updateStyleVariables({
+			[SAVED_ID_PROPERTY]: savedFeatureId,
+		});
+	};
+
+	const discardSavedFeature = (): void => {
+		savedFeature.value = undefined;
+		featuresVectorLayer.value?.updateStyleVariables({ [SAVED_ID_PROPERTY]: '' });
+	};
+
+	const isSelectedFeatureSaved = (): boolean => {
+		const savedId = savedFeature.value?.get(FEATURE_ID_PROPERTY) as string;
+		const selectedId = selectedFeature.value?.get(FEATURE_ID_PROPERTY) as string;
+		return savedId?.length > 0 && savedId === selectedId;
+	};
+
+	watch(
+		() => currentState.value,
+		() => {
+			if (currentState.value !== STATES.ERROR) {
+				errorMessage.value = undefined;
+			}
+		},
+		{ immediate: true }
+	);
+
+	return {
```

Comment:
@matthew-white this is the core of map and OL. Additional features might be added as a separate module and then imported there. For example:
```
return {
        initializeMap,
		loadGeometries,
		errorMessage,

        ...editShapesModule, // <-- for example something like this.

       ... etc
}
```